File: iOS8_Advanced/3_1_ViewTest_ObjC/ViewTest_ObjC/ViewController.m
//
//  ViewController.m
//  ViewTest_ObjC
//
//  Created by 新居雅行 on 2014/10/01.
//  Copyright (c) 2014年 msyk.net. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <UIContentContainer>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"%s", __FUNCTION__);
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    NSLog(@"%s", __FUNCTION__);
    // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)viewDidDisappear:(BOOL)animated
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)viewDidLayoutSubviews
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)viewWillAppear:(BOOL)animated
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)viewWillDisappear:(BOOL)animated
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)viewWillLayoutSubviews
{
    NSLog(@"%s", __FUNCTION__);

}

// Try to comment out the method below
- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)willTransitionToTraitCollection:(UITraitCollection *)newCollection withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)updateViewConstraints
{
    NSLog(@"%s", __FUNCTION__);
    [super updateViewConstraints];
}

- (void)awakeFromNib    {
    NSLog(@"%s", __FUNCTION__);

}

- (void)willRotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
{
    NSLog(@"%s", __FUNCTION__);

}

- (void)willAnimateRotationToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
{
    NSLog(@"%s", __FUNCTION__);
    
}

- (void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceOrientation
{
    NSLog(@"%s", __FUNCTION__);
    
}

@end

File: iOS8_Advanced/3_1_ViewTest/ViewTest/ViewController.swift
//
//  ViewController.swift
//  2_1_ViewTest
//
//  Created by 新居雅行 on 2014/07/24.
//  Copyright (c) 2014年 msyk.net. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBAction private func tapButton() {
        let vc : UIViewController = UIViewController()
        self.view.window?.rootViewController = vc;
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
        
        let tc = self.traitCollection
        println(tc.horizontalSizeClass,
            tc.displayScale,
            tc.userInterfaceIdiom,
            tc.verticalSizeClass)
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func viewDidDisappear(animated: Bool) {
        super.viewDidDisappear(animated)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    // Try to comment out the method below
    override func viewWillTransitionToSize(size: CGSize,
        withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
            println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    
    override func willTransitionToTraitCollection(
        newCollection: (UITraitCollection!),
        withTransitionCoordinator coordinator: (UIViewControllerTransitionCoordinator!))
    {
            super.willTransitionToTraitCollection(
                newCollection, withTransitionCoordinator: coordinator)
            println((__FILE__ as String).lastPathComponent, __FUNCTION__)
            
    }
    
    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection!) {
        super.traitCollectionDidChange(previousTraitCollection)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
        
    }
    
    override func updateViewConstraints() {
        super.updateViewConstraints()
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func awakeFromNib() {
        super.updateViewConstraints()
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
        
    }

    override func willRotateToInterfaceOrientation(toInterfaceOrientation: UIInterfaceOrientation, duration d: NSTimeInterval) {
        super.willRotateToInterfaceOrientation(toInterfaceOrientation, duration: d)
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
        
    }

    override func willAnimateRotationToInterfaceOrientation(toInterfaceOrientation: UIInterfaceOrientation, duration d: NSTimeInterval) {
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
    
    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
        println((__FILE__ as String).lastPathComponent, __FUNCTION__)
    }
}

